mystery_word.py

```diff
@@ -0,0 +1,265 @@
+    many guesses are reaining"""
+    a = "Make a guess! You have {} left.".format(counter)
+    print(a)
+    return a
+
+
+def repeat_guess_check(capsletter, guess_list):
+    "Checks to see if the guessed letter has already been guessed"
+    if capsletter in guess_list:
+        print("""You've already guessed that letter.  Try another.""")
+        return "guess again"
+    else:
+        return "pass"
+
+
+def error_checking(capsletter):
+    """Checks to make sure that the guess is a) not more than one
+    letter, and b) a standard letter of the alphabet"""
+    if len(capsletter) > 1:
+        print("Please guess only one letter.")
+        return "not ok"
+    elif capsletter not in "ABCDEFGHIJKLMNOPQRSTUVWXYZ":
+        print("Please guess a letter.")
+        return "not ok"
+    else:
+        return "ok"
+
+
+def append_guess_list(guess, guess_list):
+    """Appends the guessed letter to the list of guesses"""
+    guess_list.append(guess)
+    return guess_list
+
+
+def success_text():
+    """Prints successful guess text"""
+    yay = "Good guess!\n"
+    print(yay)
+    return yay
+
+
+def failure_text():
+    """Prints unsuccessful guess text."""
+    nay = "Sorry. Bad guess.\n"
+    print(nay)
+    return nay
+
+
+def get_guess():
+    """Takes an input from the user and capitalizes it"""
+    letter = input("> ")
+    capsletter = letter.upper()
+    return capsletter
+
+
+def list_visualizer(chosen_word_list, guess_list):
+    """"Takes the list form of the word you're trying to guess and checks
+    it against the list of guesses to create a string visualization """
+    usage_list = chosen_word_list[:]
+    visual_string = ""
+    for index, letter in enumerate(usage_list):
+        if letter not in guess_list:
+            usage_list[index] = "_"
+            visual_string = ' '.join(usage_list)
+    print(visual_string)
+    return visual_string
+
+
+def guess_checker(a_guess, chosen_word_string, counter):
+    """Checks to see whether the guess is part of the chosen word,
+    and lowers the counter by one for incorrect guesses"""
+    a_guess = a_guess.upper()
+    if a_guess in chosen_word_string:
+        success_text()
+        return counter
+    else:
+        counter = counter - 1
+        failure_text()
+        return counter
+
+
+def pretty_guess_list(guess_list):
+    """reformats the guess_list to be more aesthetically pleasing"""
+    pretty_list = guess_list[:]
+    pretty_list.sort()
+    pretty_string = ' '.join(pretty_list)
+    return pretty_string
+
+
+def win_check(chosen_word_list, guess_list):
+    """Tests to see whether all of the letters in the word have been guessed"""
+    if all(item in guess_list for item in chosen_word_list):
+        return True
+    else:
+        return False
+
+
+def lose_check(counter):
+    """Tests to see whether the user has run our of guesses"""
+    if counter > 0:
+        return False
+    else:
+        return True
+
```

Comment:
The two above functions should be collapsed to just use the predicate in each if statement:

```py
def win_check(chosen_word_list, guess_list):
    """Tests to see whether all of the letters in the word have been guessed"""
    return all(item in guess_list for item in chosen_word_list)

def lose_check(counter):
    """Tests to see whether the user has run our of guesses"""
    return counter <= 0